v0.19.0 crashes as soon I open the app

I was previously using v0.18.5 which was working fine for me, now after updating the app crashes as soon as I open it. I'm unable to even get into the app to backup my data. Please help.
<!--
Oh no, a bug! It happens. Thanks for reporting an issue with NewPipe.

Use this template to notify us if you found a bug.

To make it easier for us to help you please enter detailed information below.

Please note, we only support the latest version of NewPipe and the master branch. Make sure you have that version installed. If you don't, upgrade & reproduce the problem before opening the issue. The release page (https://github.com/TeamNewPipe/NewPipe/releases/latest) is the go-to place to get this version. In order to check your app version, open the left drawer and click on "About".

P.S.: Our contribution guidelines might be a nice document to read before you fill out the report :) You can find it at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md
-->
### Version
<!-- Which version are you using? -->
- v0.19.0


### Steps to reproduce the bug
<!-- If you can't reproduce it, please try to give as many details as possible on how you think you got to the bug. -->
Steps to reproduce the behavior:
1. Open app

### Expected behavior
Open the app without it crashing.

### Actual behaviour
The app crashes as soon as I open it.

### Screenshots/Screen records
None.

### Logs
If your bug includes a crash, please head over to the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/). Copy the result. Paste it here:
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ US
* __Service:__ none
* __Version:__ 0.19.0
* __OS:__ Linux samsung/d2qsq/d2q:10/QP1A.190711.020/N975USQU2BTA9:user/release-keys 10 - 29


<details><summary><b>Crash log</b></summary><p>

```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | NOT NULL constraint failed: streams_new.uploader (code 1299 SQLITE_CONSTRAINT_NOTNULL[1299])
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.subscribers.LambdaSubscriber.onError(LambdaSubscriber.java:79)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.checkTerminated(FlowableObserveOn.java:209)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnSubscriber.runAsync(FlowableObserveOn.java:399)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:7770)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1047)
Caused by: android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: streams_new.uploader (code 1299 SQLITE_CONSTRAINT_NOTNULL[1299])
	at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
	at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:1066)
	at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:756)
	at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:66)
	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:2322)
	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:2249)
	at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.execSQL(FrameworkSQLiteDatabase.java:242)
	at org.schabi.newpipe.database.Migrations$2.migrate(Migrations.java:84)
	at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:87)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:124)
	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:493)
	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:391)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:92)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:53)
	at androidx.room.RoomDatabase.inTransaction(RoomDatabase.java:452)
	at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.java:275)
	at androidx.room.RoomDatabase.query(RoomDatabase.java:304)
	at androidx.room.util.DBUtil.query(DBUtil.java:54)
	at org.schabi.newpipe.database.feed.dao.FeedDAO_Impl$10.call(FeedDAO_Impl.java:499)
	at org.schabi.newpipe.database.feed.dao.FeedDAO_Impl$10.call(FeedDAO_Impl.java:496)
	at io.reactivex.internal.operators.maybe.MaybeFromCallable.subscribeActual(MaybeFromCallable.java:46)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.onNext(FlowableFlatMapMaybe.java:132)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnSubscriber.runAsync(FlowableObserveOn.java:407)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: streams_new.uploader (code 1299 SQLITE_CONSTRAINT_NOTNULL[1299])
	at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
	at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:1066)
	at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:756)
	at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:66)
	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:2322)
	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:2249)
	at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.execSQL(FrameworkSQLiteDatabase.java:242)
	at org.schabi.newpipe.database.Migrations$2.migrate(Migrations.java:84)
	at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:87)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:124)
	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:493)
	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:391)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:92)
	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:53)
	at androidx.room.RoomDatabase.inTransaction(RoomDatabase.java:452)
	at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.java:275)
	at androidx.room.RoomDatabase.query(RoomDatabase.java:304)
	at androidx.room.util.DBUtil.query(DBUtil.java:54)
	at org.schabi.newpipe.database.feed.dao.FeedDAO_Impl$10.call(FeedDAO_Impl.java:499)
	at org.schabi.newpipe.database.feed.dao.FeedDAO_Impl$10.call(FeedDAO_Impl.java:496)
	at io.reactivex.internal.operators.maybe.MaybeFromCallable.subscribeActual(MaybeFromCallable.java:46)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.onNext(FlowableFlatMapMaybe.java:132)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnSubscriber.runAsync(FlowableObserveOn.java:407)
	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260)
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</p></details>
<hr>


<!-- That's right, here! -->

--- a/app/src/androidTest/java/org/schabi/newpipe/database/AppDatabaseTest.kt
+++ b/app/src/androidTest/java/org/schabi/newpipe/database/AppDatabaseTest.kt
@@ -25,6 +25,9 @@ class AppDatabaseTest {
         private const val DEFAULT_DURATION = 480L
         private const val DEFAULT_UPLOADER_NAME = "Uploader Test"
         private const val DEFAULT_THUMBNAIL = "https://example.com/example.jpg"
+
+        private const val DEFAULT_SECOND_SERVICE_ID = 0
+        private const val DEFAULT_SECOND_URL = "https://www.youtube.com/watch?v=ncQU6iBn5Fc"
     }
 
     @get:Rule val testHelper = MigrationTestHelper(InstrumentationRegistry.getInstrumentation(),
@@ -45,6 +48,26 @@ class AppDatabaseTest {
                 put("uploader", DEFAULT_UPLOADER_NAME)
                 put("thumbnail_url", DEFAULT_THUMBNAIL)
             })
+            insert("streams", SQLiteDatabase.CONFLICT_FAIL, ContentValues().apply {
+                // put("uid", null)
+                put("service_id", DEFAULT_SECOND_SERVICE_ID)
+                put("url", DEFAULT_SECOND_URL)
+                // put("title", null)
+                // put("stream_type", null)
+                // put("duration", null)
+                // put("uploader", null)
+                // put("thumbnail_url", null)
+            })
+            insert("streams", SQLiteDatabase.CONFLICT_FAIL, ContentValues().apply {
+                // put("uid", null)
+                put("service_id", DEFAULT_SERVICE_ID)
+                // put("url", null)
+                // put("title", null)
+                // put("stream_type", null)
+                // put("duration", null)
+                // put("uploader", null)
+                // put("thumbnail_url", null)
+            })
             close()
         }
 
@@ -53,9 +76,11 @@ class AppDatabaseTest {
 
         val migratedDatabaseV3 = getMigratedDatabase()
         val listFromDB = migratedDatabaseV3.streamDAO().all.blockingFirst()
-        assertEquals(1, listFromDB.size)
 
-        val streamFromMigratedDatabase = listFromDB.first()
+        // Only expect 2, the one with the null url will be ignored
+        assertEquals(2, listFromDB.size)
+
+        val streamFromMigratedDatabase = listFromDB[0]
         assertEquals(DEFAULT_SERVICE_ID, streamFromMigratedDatabase.serviceId)
         assertEquals(DEFAULT_URL, streamFromMigratedDatabase.url)
         assertEquals(DEFAULT_TITLE, streamFromMigratedDatabase.title)
@@ -67,6 +92,20 @@ class AppDatabaseTest {
         assertNull(streamFromMigratedDatabase.textualUploadDate)
         assertNull(streamFromMigratedDatabase.uploadDate)
         assertNull(streamFromMigratedDatabase.isUploadDateApproximation)
+
+        val secondStreamFromMigratedDatabase = listFromDB[1]
+        assertEquals(DEFAULT_SECOND_SERVICE_ID, secondStreamFromMigratedDatabase.serviceId)
+        assertEquals(DEFAULT_SECOND_URL, secondStreamFromMigratedDatabase.url)
+        assertEquals("", secondStreamFromMigratedDatabase.title)
+        // Should fallback to VIDEO_STREAM
+        assertEquals(StreamType.VIDEO_STREAM, secondStreamFromMigratedDatabase.streamType)
+        assertEquals(0, secondStreamFromMigratedDatabase.duration)
+        assertEquals("", secondStreamFromMigratedDatabase.uploader)
+        assertEquals("", secondStreamFromMigratedDatabase.thumbnailUrl)
+        assertNull(secondStreamFromMigratedDatabase.viewCount)
+        assertNull(secondStreamFromMigratedDatabase.textualUploadDate)
+        assertNull(secondStreamFromMigratedDatabase.uploadDate)
+        assertNull(secondStreamFromMigratedDatabase.isUploadDateApproximation)
     }
 
     private fun getMigratedDatabase(): AppDatabase {

--- a/app/src/main/java/org/schabi/newpipe/database/Migrations.java
+++ b/app/src/main/java/org/schabi/newpipe/database/Migrations.java
@@ -81,8 +81,16 @@ public class Migrations {
                     " duration INTEGER NOT NULL, uploader TEXT NOT NULL, thumbnail_url TEXT, view_count INTEGER, textual_upload_date TEXT, upload_date INTEGER," +
                     " is_upload_date_approximation INTEGER)");
 
-            database.execSQL("INSERT INTO streams_new (uid, service_id, url, title, stream_type, duration, uploader, thumbnail_url, view_count, textual_upload_date, upload_date, is_upload_date_approximation)"+
-                    " SELECT uid, service_id, url, title, stream_type, duration, uploader, thumbnail_url, NULL, NULL, NULL, NULL FROM streams");
+            database.execSQL("INSERT INTO streams_new (uid, service_id, url, title, stream_type," +
+                    "duration, uploader, thumbnail_url, view_count," +
+                    "textual_upload_date, upload_date, is_upload_date_approximation) " +
+
+                    "SELECT uid, service_id, url, IFNULL(title, \"\"), IFNULL(stream_type, \"VIDEO_STREAM\")," +
+                    "IFNULL(duration, 0), IFNULL(uploader, \"\"), IFNULL(thumbnail_url, \"\"), NULL," +
+                    "NULL, NULL, NULL " +
+
+                    "FROM streams " +
+                    "WHERE url IS NOT NULL");
 
             database.execSQL("DROP TABLE streams");
             database.execSQL("ALTER TABLE streams_new RENAME TO streams");

